src/Lean/Environment.lean

```diff
@@ -612,13 +612,9 @@ unsafe def Environment.freeRegions (env : Environment) : IO Unit :=
 
 def mkModuleData (env : Environment) : IO ModuleData := do
   let pExts ← persistentEnvExtensionsRef.get
-  let entries : Array (Name × Array EnvExtensionEntry) := pExts.size.fold
-    (fun i result =>
-      let state  := (pExts.get! i).getState env
-      let exportEntriesFn := (pExts.get! i).exportEntriesFn
-      let extName    := (pExts.get! i).name
-      result.push (extName, exportEntriesFn state))
-    #[]
+  let entries := pExts.map fun pExt =>
+    let state := pExt.getState env
+    (pExt.name, pExt.exportEntriesFn state)
```

Comment:
The original code was a bit weird here; this is just a peephole simplification since I was in the area.